engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java

```diff
@@ -549,6 +509,9 @@ boolean isObjectWithMembers(Object object, InteropLibrary interop) {
     if (interop.isTime(object)) {
       return false;
     }
+    if (interop.hasHashEntries(object)) {
```

Comment:
Let's inspect members/entries of objects that claim to `hasHashEntries` - Python objects representing the `numpy` numbers don't tend to have them.

Reply:
I don't understand your comment here. `isObjectWithMembers` is a guard for `equalsInteropObjectWithMembers`. In your code, if something hasHashEntries, it will not be checked for member/entries equality.

Reply:
> The guards would have to be mutually exclusive, wouldn't they? Are they?

Frankly, I don't know. Most probably, guards are not mutually exclusive. But I still don't understand why an objectWithMembers must not have hashEntries.

> Maybe we should split the node even further (move the date related comparisions into own node, there is a lot of them) and make sure our specializations are exclusive using the enum @Specialization selector pattern.

I would rather increase the test coverage in `EqualsTests` that we already have. I know that mutually not exclusive guards are not a best practice, but I don't think it hurts in this case.

Reply:
> don't understand why an `objectWithMembers` must not have `hashEntries`

If you want objects with `hashEntries` to be always handled by [equalsHashMaps](https://github.com/enso-org/enso/blob/224622ac1817760c98e6199e5e0c8b2127ab39f0/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java#L320), then you need to prevent those objects from being handled by [equalsInteropObjectWithMembers](https://github.com/enso-org/enso/blob/224622ac1817760c98e6199e5e0c8b2127ab39f0/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java#L366C11-L366C41).

That can only be done by rejecting objects with `hashEntries` in `objectWithMembers` check.